### DISCUSSION TITLE
config position confusion

### DISCUSSION THREAD

**Discussion by Graywith** [aquasecurity/kube-bench#935] [Questions and Help]
```
[root@k8s-master home]# kubectl --version
Kubernetes v1.5.2
[root@k8s-master home]# ls /etc/kubernetes/
apiserver  config  controller-manager  kubelet  proxy  scheduler
```
I don't know how chose the parameter ```--benchmark=???```cis-1.5 or cis-1.6 or null, because 1.1.1-1.1.6 always fail, no matter cis-1.5 or cis-1.6 or no `--benchmark=`. My config is /etc/kubernetes/apiserver. kube-bench always use default  /etc/kubernetes/manifests/kube-apiserver.yaml. Do i need to change config.yaml manually? but  I am confused with defaultconf and defaultkubeconfig, Is there a instruction for config.yaml?
kube-bench should know my k8s version and get the right config positon, right?
or kube-bench doesn't support my k8s version?
```
1.1.1 Run the below command (based on the file location on your system) on the
master node.
For example, chmod 644 /etc/kubernetes/manifests/kube-apiserver.yaml

1.1.2 Run the below command (based on the file location on your system) on the master node.
For example,
chown root:root /etc/kubernetes/manifests/kube-apiserver.yaml

1.1.3 Run the below command (based on the file location on your system) on the master node.
For example,
chmod 644 /etc/kubernetes/manifests/kube-controller-manager.yaml

1.1.4 Run the below command (based on the file location on your system) on the master node.
For example,
chown root:root /etc/kubernetes/manifests/kube-controller-manager.yaml

1.1.5 Run the below command (based on the file location on your system) on the master node.
For example,
chmod 644 /etc/kubernetes/manifests/kube-scheduler.yaml

1.1.6 Run the below command (based on the file location on your system) on the master node.
For example,
chown root:root /etc/kubernetes/manifests/kube-scheduler.yaml
```

question2: `Can't find oc **command**` what do i need? install openshift ?
```
[root@k8s-master home]# kube-bench --benchmark=cis-1.6 run --targets=master --v=2
I0720 15:18:12.118602   31544 util.go:472] Checking for oc
I0720 15:18:12.118765   31544 util.go:501] Can't find oc command: exec: "oc": executable file not found in $PATH
I0720 15:18:12.118868   31544 kubernetes_version.go:36] Try to get version from Rest API
I0720 15:18:12.118910   31544 kubernetes_version.go:44] Failed loading certificate Error: open /var/run/secrets/kubernetes.io/serviceaccount/ca.crt: no such file or directory
I0720 15:18:12.381997   31544 util.go:321] Failed to query kubectl: exit status 1
I0720 15:18:12.382041   31544 util.go:322] exit status 1
I0720 15:18:12.382094   31544 util.go:341] Kubectl output: Error: unknown shorthand flag: 'o' in -o


Options:
  -c, --client=false: Client version only (no server required).
      --short=false: Print just the version number.

Usage:
  kubectl version [options]

Use "kubectl options" for a list of global command-line options (applies to all commands).

Error: unknown shorthand flag: 'o' in -o
unknown shorthand flag: 'o' in -o
I0720 15:18:12.382146   31544 util.go:347] invalid character 'E' looking for beginning of value
I0720 15:18:12.382167   31544 util.go:352] Unable to get Kubernetes version from kubectl, using default version: 1.18
I0720 15:18:12.382251   31544 common.go:351] Kubernetes version: "" to Benchmark version: "cis-1.6"
I0720 15:18:12.382278   31544 run.go:40] Checking targets [master] for cis-1.6
```

**Comment by yoavrotems** [comment]
Hey could you please provide more information about your environment? what kubectl version? 
And kube-bench is supporting auto-detection it suppose to detect the running kubernetes and run the compatible cfg. 
The oc not found just means kube-bench is not assuming its ocp env and continue to check different stuff, like kubectl and kubelet output. 

About config.yaml it seems that you are right and there isn't documentation 😮 
Long story short, it checks for the values in the list top bottom if nothing found will take default.  

So about your issue with kube-apiserver it could be that either you have the two files and then it will chose the first or that you don't have your file in the list and then it will take the default
